controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -284,6 +284,7 @@ private HttpResponse handleDELETE(Path path, HttpRequest request) {
         if (path.matches("/application/v4/tenant/{tenant}")) return deleteTenant(path.get("tenant"), request);
         if (path.matches("/application/v4/tenant/{tenant}/key")) return removeDeveloperKey(path.get("tenant"), request);
         if (path.matches("/application/v4/tenant/{tenant}/application/{application}")) return deleteApplication(path.get("tenant"), path.get("application"), request);
+        if (path.matches("/application/v4/tenant/{tenant}/application/{application}/deployment")) return removeProdAllDeployments(path.get("tenant"), path.get("application"));
```

Comment:
What about test/staging? If this indeed only applies to prod, we can perhaps use the path `/application/v4/tenant/{tenant}/application/{application}/environment/prod`?

Reply:
My understanding is that this will submit the empty application, which will notify the JobController which will asynchronously at some point see that prod deployments have been removed and start making DELETE calls against config servers. It will also figure out that the system/staging tests it might have been running are no longer needed and abort them + delete deployment (similar to what happens when you submit a new package while tests are running).